agent/http.go

```diff
@@ -231,13 +232,12 @@ func (s *HTTPServer) wrap(handler func(resp http.ResponseWriter, req *http.Reque
 
 		handleErr := func(err error) {
 			s.agent.logger.Printf("[ERR] http: Request %s %v, error: %v from=%s", req.Method, logURL, err, req.RemoteAddr)
-			code := http.StatusInternalServerError // 500
-			errMsg := err.Error()
-			if strings.Contains(errMsg, "Permission denied") || strings.Contains(errMsg, "ACL not found") {
-				code = http.StatusForbidden // 403
+			switch {
+			case acl.IsErrPermissionDenied(err) || acl.IsErrNotFound(err):
+				http.Error(resp, err.Error(), http.StatusForbidden) // 403
+			default:
+				http.Error(resp, err.Error(), http.StatusInternalServerError) // 500
 			}
-			resp.WriteHeader(code)
```

Comment:
Since we're using the `resp.WriteHeader(...); fmt.Fprint(resp, ...)` pattern everywhere else I've reverted that. Consistency is more important and this also has the potential to break someone's response parsing. I think the additional header should not have that effect.